python/tank/authentication/console_authentication.py

```diff
@@ -47,7 +49,7 @@ def authenticate(self, hostname, login, http_proxy):
         :param hostname: Host to renew a token for.
         :param login: User to renew a token for.
         :param http_proxy: Proxy to use for the request. Can be None.
-        :returns: The (hostname, login, session token) tuple.
+        :returns: The (hostname, login, session token, cookies) tuple.
```

Comment:
would be good with more info about what kind of data structure cookies is.

Reply:
The cookies is meant to be an opaque structure, not meant for user consumption (ironic with a name like cookies :-)  )

So I guess another name would be more suited...  I have used 'encoded_cookies' in some places, but I do not think it is well suited either.

One thing I have been thinking about is using something like 'sso_context' or 'session_context'.  Which could be a dictionary with a key 'sso_cookies', and could be expanded with other items should it be needed.

Again, I want a name that suggests 'no serviceable part inside'

Reply:
oh i like `session_context`!

Reply:
In that spririt... Should session_context be a dictionary with keys `hostname`, `login`, `session_token` and (optionally) `sso_cookies` ?

That would tie in with my refactoring that I was eluding to in my PR description. I would actually create a class SessionContext, with proper getters/setters.

Reply:
sorry to confuse things - i think perhaps for now, it might be overkill to do that full refactor that you outline (even though i really like the idea!). I misunderstood (and didn't read it properly), thinking you meant we just rename `cookies` -> `session_context `. I think the point i was trying to make was that `cookies` is perhaps more describing how it's stored than that is is - wondering if something like `session_metadata` is a more appropriate name than `cookies` or `sso_cookies` if we are aiming for a generic black box property bag where we may end up storing more stuff in the future.

Reply:
I'll go with session_context.
I'll stick to a simple string replace in the public interface and comments then.